carbon-design-system/carbon
# feat(expressive): add expressive variation for the accordion and button

Closes: 
[Create RFC for an expressive moment in a Carbon core component #1782](https://github.com/carbon-design-system/ibm-dotcom-library/issues/1782)

#### Description
This is an RFC for an approach the IBM.com team has suggested to introduce the expressive theming. This approach creates an expressive variant of a carbon component that would users to set the expressive theming per component.

Previously the IBM.com team created a proxy that would allow for expressive theming to be applied to carbon's core components when a flag is turned on. A major issue with this approach was the maintainability - every time the carbon dependencies were updated in the IBM.com repo we would have to comb through all components and ensure the expressive styles were still intact. 

With this RFC approach, it would be easier to maintain the expressive styles as the styles are built into the core component itself. 

#### Changelog

**New**

- `expressive` prop to be passed to accordion and set a `--accordion--expressive` modifier classname
- add expressive styles under the new modifier class
- `expressive` knob for storybook accordion story

#### Testing / Reviewing

![Apr-02-2020 10-50-51](https://user-images.githubusercontent.com/54281166/78263511-0db1d800-74d0-11ea-9893-7487af06b572.gif)

Steps to Review:
1. go to Accordion story on storybook
2. check the `expressive theme` knob
3. Notice the expressive styling changes that appear
